File: java/04_FindMedianSortedArray.java
// https://leetcode.com/problems/median-of-two-sorted-arrays/
public double findMedianSortedArrays(int[] nums1, int[] nums2) {
    int[] mergedArray = new int[nums1.length + nums2.length];
    System.arraycopy(nums1, 0, mergedArray, 0, nums1.length);
    System.arraycopy(nums2, 0, mergedArray, nums1.length, nums2.length);
    Arrays.sort(mergedArray);
    int middle = mergedArray.length / 2;
    if (mergedArray.length % 2 == 0) {
        return (mergedArray[middle] + mergedArray[middle-1]) / 2.0;
    } else {
        return mergedArray[middle];
    }
}

File: swift/04_FindMedianSortedArray.swift
// https://leetcode.com/problems/median-of-two-sorted-arrays/
func findMedianSortedArrays(_ nums1: [Int], _ nums2: [Int]) -> Double {
    let mergedArr = (nums1 + nums2).sorted()
    let middle = mergedArr.count / 2
    if mergedArr.count % 2 == 0 {
        return (Double(mergedArr[middle]) + Double(mergedArr[middle - 1])) / 2
    } else {
        return Double(mergedArr[middle])
    }
}
